### DISCUSSION TITLE
How to test icons local ?

### DISCUSSION THREAD

**Discussion by adsingh14** [simple-icons/simple-icons#6689] [Help with SVGs] [answered]
Hi ! I've created an icon but unable to test it locally. I tried both docker and npm way.

> Build and test the package using `$ npm test` 
: This command working

>Run the project linting process using `$ npm run lint` 
: This command showed several errors after passing icon test.

**TL;DR:**
![image](https://user-images.githubusercontent.com/35252877/136614524-2d637bfd-d2e0-46ba-8c39-c4bf6e7a0804.png)
![image](https://user-images.githubusercontent.com/35252877/136613460-1d22885f-31ab-4ecc-82f5-4c9c55dfb262.png)

In docker,
First command runs, `docker build . -t simple-icons` create a container successfully
`docker run -it --rm --entrypoint "/bin/ash" simple-icons` this command doesn't run.

I prefer the npm way but why it creates an error if I followed the rules?

**Comment by jorgeamadosoria** [comment]
you said you created an icon: which one? did you made a branch or submitted a PR? If you did please send me the link to see if I can reproduce the errors you report.

I'm going to focus on the npm way, and ignore docker for the time being.

**Reply by adsingh14** [reply]
PR link #6692

**Reply by adsingh14** [reply]
I analyzed that my system shows a newline which have not created in repo.

**Reply by jorgeamadosoria** [reply]
![image](https://user-images.githubusercontent.com/15719479/136669190-6b2b374d-5f1b-46ff-bd69-d6589c855800.png)

No errors here (this is the output of `npm run lint`)
Maybe you need to execute `npm install `again?

Please try to do that, then execute `npm run lint` (skip `npm test`) and let me know what output you get.

**Reply by adsingh14** [reply]
Okay. I'll get back after re-installing `np`.

**Reply by adsingh14** [reply]
Still getting same error. I should clone the repo again.

**Reply by jorgeamadosoria** [reply]
The problem you get is related to trailing new lines only?
do you have an empty line at the end of svg.config.js,icons.test.js, index.test.js and simple-icons.json?
You should have it. If you do, and you still get this error, then maybe the problem is with the editor? What do you use, vscode?

**Reply by adsingh14** [reply]
Yes, VSCode. I found the problem but it occurs, even I pulled fresh repo and avoid to open it in any IDE.
Before pull, I checked the feature to remove newlines in VSCode settings.

![image](https://user-images.githubusercontent.com/35252877/136669816-1b4cb5f0-c749-4161-ab57-c39457d6d8fd.png)

**Reply by adsingh14** [reply]
Any suggestion ?

**Reply by jorgeamadosoria** [reply]
well, my problem is different:
When I first cloned the repository, the problem I had was that every time I saved the simple-icons.json file (and actually, any json file) VSCode would delete the last newline. 
I went around that by assigning a shortcut to saving without formatting (see first result here: https://www.google.com/search?q=save+without+format+vscode&rlz=1C5CHFA_enMX919MX919&oq=save+without+format+vscode&aqs=chrome..69i57.4811j0j1&sourceid=chrome&ie=UTF-8).

But that may not be of use to you.

**Reply by adsingh14** [reply]
Unfortunately,  this is not useful to me. I pulled original repo and that shows the same error on local. But I wonder why my PR doesn't have those `newlines`. In that case my PR could be merged by Maintainer, though ?

**Reply by jorgeamadosoria** [reply]
> Unfortunately, this is not useful to me. I pulled original repo and that shows the same error on local. But I wonder why my PR doesn't have those `newlines`. In that case my PR could be merged by Maintainer, though ?

Probably yes, I see no reason why not. I'll check it and merge it if the icon is fine. 

I still think the problem is with vs code configuration. Do you have Prettier installed?

**Reply by adsingh14** [reply]
Hi there, I disabled every extension and changed the CRLF to LF. This issue has generated from git config not from IDE. 
I'll update my branch after fixing it.

**Reply by adsingh14** [reply]
I think that's something with Windows git bash editor. I tried the same on clean install Windows PC and unable to run that but I commit and PR by rules.

**Reply by jorgeamadosoria** [reply]
Well, your second PR looks good, I already merged it.
You are welcome to keep contributions going, I'm sure if you keep at it you will find the problem that is not letting you run the project.

**Reply by adsingh14** [reply]
Well, I'm doing 'research' 😅 on this. I'll add more icons and happy to maintain this repo as well.

I will try another ways to run this project locally. May I know which OS you're using?

**Reply by jorgeamadosoria** [reply]
Currently I'm on MacOS. It's not my system of choice, but it seems to work just fine for the project.

**Reply by adsingh14** [reply]
Looks like Mac or Linux, both are good for non-Windows development.
I tried to run lint command on 'zsh', still same issue. 

Nodejs: v16

**Reply by jorgeamadosoria** [reply]
my node is 14.15
Windows may be problematic for the command lines, yes.

**Comment by ericcornelissen** [comment]
@adsingh14 this can indeed be a problem on Windows due to the fact that Windows uses different characters for line-endings than Unix (Linux and Mac) systems. This is enforced to avoid "phantom" changes (where a file has changed in the line-ending it has, which is not visible).

Git for Windows has an option for this that you have to configure correctly for this problem to go away. If I'm not mistaking you can run `git config --global core.autocrlf true`, possible re-clone this repository, and try again. If that doesn't work, researching this option (`core.autocrlf`) should point you in the right direction to fix the problem.

**Comment by adsingh14** [comment, marked as answer]
From recent changes. The CRLF error has resolved! Even it's working fine on 'bash' as well as 'Ubuntu Terminal'.

![image](https://user-images.githubusercontent.com/35252877/141166837-a2c01aea-d7ac-4b3a-878c-17a92f32b010.png)
